src/methods/Method_H.cpp

```diff
@@ -0,0 +1,158 @@
+#include "../Analysis.h"
+#include "../general/Logger.h"
+#include "Method_H.h"
+#include "../general/MethodFunctions.h"
+
+#include <fstream>
+#include <iostream>
+
+using std::ofstream;
+using std::string;
+using std::vector;
+
+Method_H::Method_H()
+{
+    _dx              = 0;
+    _minFrame        = 0;
+    _deltaT          = -1;
+    _fps             = 16;
+    _areaForMethod_H = nullptr;
+    _numPeds         = 0;
+}
+
+Method_H::~Method_H() = default;
+
+bool Method_H::Process(const PedData & peddata)
+{
+    _trajName       = peddata.GetTrajName();
+    _projectRootDir = peddata.GetProjectRootDir();
+    _outputLocation = peddata.GetOutputLocation();
+    _peds_t         = peddata.GetPedIDsByFrameNr();
+    _numPeds        = peddata.GetNumPeds();
+    _xCor           = peddata.GetXCor();
+    _yCor           = peddata.GetYCor();
+    _minFrame       = peddata.GetMinFrame();
+    _fps            = peddata.GetFps();
+    _firstFrame     = peddata.GetFirstFrame();
+    _tIn            = std::vector<int>(_numPeds, 0);
+    _tOut           = std::vector<int>(_numPeds, 0);
+    _entrancePoint  = std::vector<point_2d>(_numPeds, boost::geometry::make<point_2d>(0, 0));
+    _exitPoint      = std::vector<point_2d>(_numPeds, boost::geometry::make<point_2d>(0, 0));
+    if(_deltaT == -1) {
+        _deltaT = peddata.GetNumFrames();
+    }
+
+    _measureAreaId           = boost::lexical_cast<string>(_areaForMethod_H->_id);
+    std::ofstream fRhoVFlow =
+        GetFile("flow_rho_v", _measureAreaId, _outputLocation, _trajName, "Method_H");
+    if(!fRhoVFlow.is_open()) {
+        LOG_ERROR("Cannot open file to write density, flow and velocity data for method H!\n");
+        exit(EXIT_FAILURE);
+    }
+    fRhoVFlow << "#mean flow (1 / s)\tmean density (1 / m)\tmean velocity (m / s)\n";
+
+    LOG_INFO("------------------------Analyzing with Method H-----------------------------");
+    GetTinToutEntExt(peddata.GetNumFrames());
+    
+    if(_areaForMethod_H->_length < 0) {
+        LOG_WARNING("The measurement area length for method H is not assigned! Cannot calculate "
+                    "mean density, velocity and flow!");
+        exit(EXIT_FAILURE);
+    } else {
+        _dx = _areaForMethod_H->_length;
+        LOG_INFO("The measurement area length for method H is {:.3f}", _areaForMethod_H->_length);
+    }
+    OutputRhoVFlow(peddata.GetNumFrames(), fRhoVFlow);
+    fRhoVFlow.close();
+
+    return true;
+}
+
+void Method_H::GetTinToutEntExt(int numFrames)
+{
+    vector<bool> IsinMeasurezone(_numPeds, false);
+
+    for(int frameNr = 0; frameNr < numFrames; frameNr++) {
+        vector<int> ids       = _peds_t[frameNr];
+        int pedsinMeasureArea = 0;
+        for(int ID : ids) {
+            int x = _xCor(ID, frameNr);
+            int y = _yCor(ID, frameNr);
+
+            int nextX, nextY;
+            if((frameNr + 1) < numFrames) {
+                nextX = _xCor(ID, frameNr + 1);
+                nextY = _yCor(ID, frameNr + 1);
```

Comment:
Usually it is better to initialize variables directly with the desired values, otherwise they may store some strange values. Also try to declare variables as late as possible such that they are only accessible in the scope where they are needed.

```suggestion
            if((frameNr + 1) < numFrames) {
                int nextX = _xCor(ID, frameNr + 1);
                int nextY = _yCor(ID, frameNr + 1);
```